src/main/java/com/example/springboot/controller/CategoriesController.java

```diff
@@ -0,0 +1,72 @@
+package com.example.springboot.controller;
+
+import com.example.springboot.dto.BookDtoWithoutCategoryIds;
+import com.example.springboot.dto.CategoryDto;
+import com.example.springboot.dto.CreateCategoryRequestDto;
+import com.example.springboot.service.BookService;
+import com.example.springboot.service.CategoryService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import java.util.List;
+import javax.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Category management", description = "Endpoints for managing category")
+@RequiredArgsConstructor
+@RestController
+@RequestMapping(value = "/categories")
+public class CategoriesController {
```

Comment:
dont use plural form
```suggestion
public class CategoryController {
```